### DISCUSSION TITLE
【任务（已完成）】“审核+图片补充”计划 1｜20,000 $PEOPLE

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#18] [任务 | Bounty]
目前WTFSolidity极简教程已经有接近30讲，但仍然没有经过严格的编辑审核，有的部分也少一些概念的示意图或程序运行的截图，不利于新手学习。
为了让WTFSolidity极简教程更好，我发起“审核+图片补充”计划，每一讲由1个社区成员认领，负责审核校对，并添加一些图片。
**赏金**：20,000 $PEOPLE 代币，根据工作量分配。
在这个帖子下面回复就能认领（目前1人最多认领5讲，完成后可认领新的）。
有用的资料：
1. Markdown书写规则：https://www.markdownguide.org/cheat-sheet/
2. 审核并添加图片后的例子：https://github.com/AmazingAng/WTFSolidity/tree/main/02_ValueTypes

认领情况：
第1讲：三行代码HelloWeb3
0xAA **已完成**

第2讲：数值类型
@Lokiscripter  **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/22)**

第3讲：函数类型 (external/internal/public/private, pure/view/payable)
@Lokiscripter **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/44)**

第4讲：变量作用域和数据存储 (storage/memory/calldata)
@Lokiscripter **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/44)**

第5讲：数组（array）和结构体（struct）
@Lokiscripter **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/44)**

第6讲：映射（mapping）
@lcy101u **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/36)**

第7讲：控制流（control flow）和插入排序（insertion sort）
@lcy101u **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/37)**

第8讲：构造函数（constructor）和修饰器（modifier）
@wishucry **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/26)**

第9讲：事件（events）
@wishucry **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/26)**

第10讲：继承(inheritance)
@Eanam **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/30)**

第11讲：抽象合约(abstract)和接口(interface)
@Eanam **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/35)**

第12讲：库合约(library)
@wishucry **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/26)**

第13讲：异常(errors)
@Rulesbreaker **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/25)**

第14讲：常量(constant/immutable)
@Rulesbreaker **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/40)**

第15讲：函数重载(overloading)
@wishucry **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/26)**

第16讲：接收ETH(fallback/receive)
@wishucry **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/26)**

第17讲：发送ETH(transfer/send/call)
@Hongchenglong **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/29)**

第18讲：函数输出(returns/return)
@HuiTaiLa **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/38)**

第19讲：调用其他合约
@XiaoYao-0  **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/31)**

第20讲：变量初始值
@HuiTaiLa **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/38)**

第21讲：Call
@Hongchenglong **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/39)**

第22讲：Delegatecall
@xxycfhb **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/43)**

第23讲：在合约中创建新合约
@jie1789  **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/23)**

第24讲：Create2
@tangminjie **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/19)**

第25讲：删除合约
@tangminjie **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/24)**

第26讲：ABI编码解码
@HuiTaiLa **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/41)**

第27讲：Hash
@HuiTaiLa **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/41)**

第28讲：函数选择器（selector）
@XiaoYao-0 **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/27)**

第29讲：Try-Catch
@jie1789 **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/73)**

第30讲：Import
@JustinAsdz **已完成，[PR](https://github.com/AmazingAng/WTFSolidity/pull/42)**

交叉审核
讲   审核人
1	@wishucry 
2	@wishucry 
3	@wishucry 
4	@wishucry 
5	@wishucry 
6	@Rulesbreaker (完成)
7	@Rulesbreaker (完成)
8	@Lokiscripter
9	@Lokiscripter
10	@lcy101u 
11	@lcy101u 
12	@Lokiscripter
13	@HuiTaiLa
14	@HuiTaiLa
15	0xAA
16	@Lokiscripter
17	@Eanam
18	@XiaoYao-0
19	@Hongchenglong
20	@XiaoYao-0
21	@Eanam
22	@jie1789
23	@xxycfhb 
24	@HuiTaiLa (完成)
25	@HuiTaiLa
26	@tangminjie (完成)
27	@tangminjie 
28	@Hongchenglong
29	@JustinAsdz (完成)
30	@jie1789

WTF Solidity极简教程第一次审核赏金，共 20,000 $PEOPLE 发放完毕，由我个人出资。学solidity赚 $PEOPLE 听起来不错。
感谢12位贡献者的付出！What's next?
开源教程链接：https://github.com/AmazingAng/WTFSolidity

交易hash：https://bscscan.com/tx/0xcf66e73cd4adaeade00e4e23edb38f5d130e5c35d94fa4ed5a02f7720d19d99f

**Comment by Lokiscripter** [comment]
我认领 第1讲 to 第5讲

**Comment by Eanam** [comment]
认领 
- [ ] 第十讲（继承） 
- [ ] 第十一讲（抽象合约和接口合约）

**Comment by lcy101u** [comment]
認領
第6讲：映射（mapping）

第7讲：控制流（control flow）和插入排序（insertion sort）

**Comment by tangminjie** [comment]
认领 
第24讲：Create2

第25讲：删除合约

**Comment by Rulesbreaker** [comment]
认领
第13讲：异常(errors)
第14讲：常量(constant/immutable)

**Comment by wishucry** [comment]
认领 8，9，12，15，16，17，19，21，22，28

**Reply by AmazingAng** [reply]
目前1人最多认领5讲，完成后可认领新的
算你认领8，9，12，15，16

**Reply by wishucry** [reply]
好

**Comment by HuiTaiLa** [comment]
认领18、20、26、27

**Comment by XiaoYao-0** [comment]
认领19，28

**Comment by Hongchenglong** [comment]
认领17，19，21

**Comment by xxycfhb** [comment]
认领22

**Comment by jie1789** [comment]
认领23

**Comment by JustinAsdz** [comment]
认领完了😂

**Reply by AmazingAng** [reply]
哈哈。。没事，等新的！

**Comment by JustinAsdz** [comment]
～～认领10，11～～
第10讲：继承(inheritance)
第11讲：抽象合约(abstract)和接口(interface)

**Comment by jie1789** [comment]
认领29

**Comment by JustinAsdz** [comment]
认领30

**Reply by AmazingAng** [reply]
好

**Comment by AmazingAng** [comment]
赏金更新：

Github id | 审核哪些讲 | 审核总数 | 钱包地址 | 赏金 （$PEOPLE）
-- | -- | -- | -- | --
@AmazingAng | 1 | 1 | 0x3317AD9eDa6942b5a7BE5BA83346C0Ea82C3C26C | 666
@tangminjie | 24,25 | 2 | 0xcf3285e093cd93015e7456911e2db34AB4975bAd | 1333
@xxycfhb | 22 | 1 | 0xa10751935ce8A8c4678ecf616BaA0f2dfc001EC6 | 666
@Rulesbreaker | 13,14 | 2 | 0x7fe42Fbb75320a109cdc9ffd32ea801fA570d626 | 1333
@XiaoYao-0 | 19,28 | 2 | 0x4af58C88B28469F6Cff1D9d74fa2E0D5A84B389A | 1333
@JustinAsdz | 30 | 1 | 0x3Da4da2De7a937aEbBe26166B4f06EDE7Bc77BE5 | 666
@lcy101u | 6,7 | 2 | 0xe78a1782F4Cf881fa326C04b47a6f8846aaa94b7 | 1333
@HuiTaiLa | 18,20,26,27 | 4 | 0xBF06846aF2321372a273C3465DD2B3bc754A7E9d | 2666
@Lokiscripter | 2,3,4,5 | 4 | 0x22d20517e3Ae77fb666bb4ea4248064B88C57285 | 2666
@Hongchenglong | 17,21 | 2 | 0x034eAbd89eF477e413e8ded4916F6B22271Ec022 | 1333
@Eanam | 10,11 | 2 | 0x1da8017F44D3a204EeE18CBeec15cDcc7bC127F7 | 1333
@wishucry | 8,9,12,15,16 | 5 | 0x777777775a09c9aD5b63B62918023272389Fa9ce | 3333
@jie1789 | 23,29 | 2 | 0x53EFe7c1557DcdA3e686764ef36178924863080B | 1333

**Comment by JustinAsdz** [comment]
33还审吗 预约下～
